### DISCUSSION TITLE
add on/off options for icon

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#13700] [Dashboard]
HA has come a long way in using icons for various domains and device_class: allowing to define icons for various predefined states. On/Off, home/not_home among the most obvious.

We can set an icon in customize for an entity, and UI configuration has improved enormously with the icon_picker.

However, we can only set a fixed icon, if not using a device_class, and, even then, the device_class uses a fixed set of icon options.

Material design icons offer a huge amount of -off icons. With this discussion I'd love to suggest to add _a core option for icons in regular and off state_. It would tremendously add to the visual impact and meaningfulness of the frontend seeing the state of various entities immediately.

I've been doing that since forever with [custom-ui](https://github.com/Mariusthvdb/custom-ui/blob/master/README.md), which allows for templates on the attributes  icon being the most used of those.
Rather do such basic things in core though, and not have to use a custom resource for that.

example:

```
homeassistant:
  customize_glob:

     switch.cv_*:
         icon: >
           return state === 'on' ? 'mdi:radiator' : 'mdi:radiator-off';

     switch.freezer_*:
       templates:
         icon: >
           return state === 'on' ? 'mdi:fridge' : 'mdi:fridge-off';

```

<img width="257" alt="Schermafbeelding 2022-09-12 om 15 41 53" src="https://user-images.githubusercontent.com/33354141/189669539-13f2bc1a-946b-44af-b3cf-d03f740a7b4f.png">

I am aware we can do things like this with the custom cards, as the button-card above shows, but it would be so much better if we could also do this with regular entities in type:entities card which now show like:

<img width="489" alt="Schermafbeelding 2022-09-12 om 15 44 11" src="https://user-images.githubusercontent.com/33354141/189670076-34474949-322d-4b59-8546-4dc63f2e3096.png">
<img width="480" alt="Schermafbeelding 2022-09-12 om 15 44 22" src="https://user-images.githubusercontent.com/33354141/189670083-baefca03-ecf1-420b-b842-81fea61e0159.png">

instead of:

<img width="491" alt="Schermafbeelding 2022-09-12 om 15 45 41" src="https://user-images.githubusercontent.com/33354141/189670359-0b82c8d8-0129-4014-932c-47e2fc6dfbab.png">

so, please consider adding a core option for setting sate dependent icons for entities to the icon picker:

<img width="557" alt="Schermafbeelding 2022-09-12 om 15 47 46" src="https://user-images.githubusercontent.com/33354141/189670836-483cc183-43a8-4b5a-b1b2-7cf0169efcdb.png">

**Comment by bcutter** [comment]
That's exactly my current "number one feature I need to have" for the frontend 👍🏼

**Comment by Mariusthvdb** [comment]
figured this would be low hanging fruit for the WTH: https://community.home-assistant.io/t/wth-why-cant-we-add-an-on-and-off-icon-in-the-ui/467241

**Comment by Mariusthvdb** [comment]
half a year passed, many color options are added.
please have a look if this would be an option core frontend could implement? it would really complement the new color schemes perfectly, for those entities not belonging to a device_class.

a simple option in the frontend. Instead of the current single icon option:

<img width="579" alt="Scherm­afbeelding 2023-03-11 om 22 40 54" src="https://user-images.githubusercontent.com/33354141/224512672-345fd745-a503-4c8c-9f38-919d7f9f3098.png">

add an option for 

Off-icon:
On-icon:

please consider, thanks!

**Comment by Mariusthvdb** [comment]
make that  a happy BD... 

no response at all, which really is unfortunate.
